python/grass/jupyter/interact_display.py

```diff
@@ -0,0 +1,163 @@
+        center = gs.parse_command("g.region", flags="cg", env=self._vector_env)
+        center = (float(center["center_northing"]), float(center["center_easting"]))
+
+        # Create Folium Map
+        self.map = folium.Map(
+            width=self.width,
+            height=self.height,
+            location=center,
+            tiles="cartodbpositron",
+        )
+
+    def _convert_coordinates(self, x, y, proj_in):
+        """This function reprojects coordinates to WGS84, the required
+        projection for folium.
+
+        Coordinates should be a string with x and y values separated
+        by a comma.
+
+        proj_in is a proj4 string, for example, the output of g.region
+        with the `g` flag."""
+        # Reformat input
+        coordinates = f"{x}, {y}"
+
+        # Reproject coordinates
+        coords_folium = gs.read_command(
+            "m.proj",
+            coordinates=coordinates,
+            proj_in=proj_in,
+            separator="comma",
+            flags="do",
+        )
+
+        # Reformat from string to array
+        coords_folium = coords_folium.strip()  # Remove '\n' at end of string
+        coords_folium = coords_folium.split(",")  # Split on comma
+        coords_folium = [float(value) for value in coords_folium]  # Convert to floats
+
+        return coords_folium[1], coords_folium[0]  # Return Lat and Lon
+
+    def _convert_extent(self, env=None):
+        """This function returns the bounding box of the current region
+        in WGS84, the required projection for folium"""
+
+        # Get proj of current GRASS region
+        proj = gs.read_command("g.proj", flags="jf", env=env)
+
+        # Get extent
+        extent = gs.parse_command("g.region", flags="g", env=env)
+
+        # Convert extent to EPSG:3857, required projection for Folium
+        north, east = self._convert_coordinates(extent["e"], extent["n"], proj)
+        south, west = self._convert_coordinates(extent["w"], extent["s"], proj)
+
+        extent = {"north": north, "south": south, "east": east, "west": west}
+
+        return extent
+
+    def _folium_bounding_box(self, extent):
+        """Reformats extent into bounding box to pass to folium"""
+        return [[extent["north"], extent["west"]], [extent["south"], extent["east"]]]
+
+    def add_vector(self, name):
+        """Imports vector into temporary WGS84 location,
+        re-formats to a GeoJSON and adds to folium map"""
+
+        # Find full name of vector
+        full_name = gs.find_file(name, element="vector")["fullname"]
+        name = gs.find_file(name, element="vector")["name"]
+        # Reproject vector into WGS84 Location
+        gs.run_command(
+            "v.proj",
+            input=full_name,
+            location=gs.gisenv(env=os.environ)["LOCATION_NAME"],
+            dbase=gs.gisenv(env=os.environ)["GISDBASE"],
+            env=self._vector_env,
+        )
+
+        # Convert to GeoJSON
+        json_file = f"{self.tmp_dir}/tmp_{name}.json"
```

Comment:
Use Path from pathlib to manipulate the paths or os.path.join, i.e., do not construct paths using string formatting (filename and directory names yes, but not paths).

Assuming that self.tmp_dir is already a Path, the statement will look like:

```
        json_file = self.tmp_dir / f"tmp_{name}.json"
```

The disadvantage of Path, is that you need to convert it to string, but things like format strings work out of the box:

```
>>> from pathlib import Path
>>> a = Path("/tmp")
>>> f"{a}"
'/tmp'
```